### DISCUSSION TITLE
Phase-out of "renovate" npm user for private packages

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#20613] [Request Help]
Very long ago, the first way to support private packages on npmjs.org was to invite the `renovate` npm user into the user's npm org. The Renovate app would then run with the credentials from this user pre-populated and be able to perform lookups.

We have long deprecated this approach for security reasons, but some early app users are still using it. We know who these github accounts are, but don't have an easy way to reach out to them, which makes it hard to switch this feature off.

I am going to begin a "brown out" where the app will intentionally skip provisioning this npm token 20% of the time. For those still relying on it, I anticipate that they might see some autoclosed PRs, which will then be reopened the next time Renovate runs with the token. I'm hoping they'll see this enough to come ask questions, and can eventually find this Discussion.

In short, they'll need to have their own service user token and configure it like this:

```json
{
  "hostRules": [
    {
      "hostType": "npm",
      "matchHost": "https://registry.npmjs.org",
      "encrypted": {
        "token": "3f832f2983yf89hsd98ahadsjfasdfjaslf............"
      }
    }
  ]
}
```

More details here: https://docs.renovatebot.com/getting-started/private-packages/#encrypting-secrets

Suggestions welcome, naturally. Feel free to reach out to me directly via email too.

**Comment by HonkingGoose** [comment]
@rarkins Do you want to create a announcement bar on the docs repository, to explain the brownouts, and what users should do instead?

**Reply by rarkins** [reply]
Great idea!

**Reply by HonkingGoose** [reply]
@rarkins I'll go write a draft PR for you to comment on then. 😉

**Reply by HonkingGoose** [reply]
Here's the PR that enables the announcement bar and the message:

- https://github.com/renovatebot/renovatebot.github.io/pull/273

**Comment by lantrix** [comment]
Thanks! This got our attention.
